File: 2020/js/day1.js
module.exports = ( input ) =>
{
	input = input
		.split( '\n' )
		.map( x => parseInt( x, 10 ) );

	let part1 = 0;
	let part2 = 0;

	for( let a = 0; a < input.length; a++ )
	{
		for( let b = a + 1; b < input.length; b++ )
		{
			if( !part1 && input[ a ] + input[ b ] === 2020 )
			{
				part1 = input[ a ] * input[ b ];
			}

			for( let c = b + 1; c < input.length; c++ )
			{
				if( input[ a ] + input[ b ] + input[ c ] === 2020 )
				{
					part2 = input[ a ] * input[ b ] * input[ c ];
				}
			}
		}
	}

	return [ part1, part2 ];
};

File: 2025/Answers/Solutions/Day1.cs
using System;

namespace AdventOfCode;

[Answer(1)]
public class Day1 : IAnswer
{
	public Solution Solve(string input)
	{
		var part1 = 0;
		var part2 = 0;
		var dial = 50;

		foreach (var line in input.AsSpan().EnumerateLines())
		{
			var value = line[1..].ParseInt();

			if (line[0] == 'L')
			{
				if (dial == 0)
				{
					dial = 100 - value;
				}
				else
				{
					dial -= value;
				}

				if (dial < 0)
				{
					var crossings = (-dial + 99) / 100;
					part2 += crossings;
					dial += crossings * 100;
				}

				if (dial == 0)
				{
					part2++;
				}
			}
			else
			{
				dial += value;

				if (dial > 99)
				{
					var (quotient, remainder) = Math.DivRem(dial, 100);
					part2 += quotient;
					dial = remainder;
				}
			}

			if (dial == 0)
			{
				part1++;
			}
		}

		return new(part1.ToString(), part2.ToString());
	}
}
